CMakeLists.txt

```diff
@@ -1,330 +1,712 @@
-# and install it
-set (NLOPT_LIBRARIES ${nlopt_lib})
-
-set (NLOPT_CMAKE_DIR "${INSTALL_CMAKE_DIR}")
-file (RELATIVE_PATH rel_include_dir "${NLOPT_CMAKE_DIR}" "${INSTALL_INCLUDE_DIR}")
-list (APPEND RELATIVE_NLOPT_INCLUDE_DIRS ${rel_include_dir})
-
-file (RELATIVE_PATH rel_lib_dir "${NLOPT_CMAKE_DIR}" "${INSTALL_LIB_DIR}")
-list (APPEND RELATIVE_NLOPT_LIB_DIR ${rel_lib_dir})
-
-configure_file (cmake/NLoptConfig.cmake.in NLoptConfig.cmake @ONLY)
-configure_file (cmake/NLoptConfigVersion.cmake.in NLoptConfigVersion.cmake @ONLY)
-install (FILES
-          ${CMAKE_CURRENT_BINARY_DIR}/NLoptConfig.cmake
-          ${CMAKE_CURRENT_BINARY_DIR}/NLoptConfigVersion.cmake
-         DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR}
-         COMPONENT Development)
-
+#==============================================================================
+# NLOPT CMake file
+#
+# NLopt is a free/open-source library for nonlinear optimization, providing
+# a common interface for a number of different free optimization routines
+# available online as well as original implementations of various other
+# algorithms
+# WEBSITE: http://ab-initio.mit.edu/wiki/index.php/NLopt
+# AUTHOR: Steven G. Johnson
+#
+# This CMakeLists.txt file was created to compile NLOPT with the CMAKE utility.
+# Benoit Scherrer, 2010 CRL, Harvard Medical School
+# Copyright (c) 2008-2009 Children's Hospital Boston
+#==============================================================================
+
+cmake_minimum_required (VERSION 2.8.9)
+
+# default install directory locally inside build-tree
+# (this must come before calling PROJECT)
+if (NOT DEFINED CMAKE_INSTALL_PREFIX
+    OR CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set (CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/install"
+    CACHE PATH "Installation directory prefix" FORCE)
+endif ()
+
+# default build type to optimized release
+if (NOT DEFINED CMAKE_BUILD_TYPE)
+  set (CMAKE_BUILD_TYPE "Release"
+    CACHE STRING "Build type: Debug|Release|RelWithDebInfo|MinSizeRel")
+endif ()
+
+# C language project
+project (nlopt C)
+
+# append path to custom cmake modules FindXXX
+list (APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
+# provide user options
+option (BUILD_SHARED_LIBS       "Build NLopt as a shared library" ON)
+option (WITH_CXX                "Enable C++-based routines (StoGO)" OFF)
+option (WITH_SWIG               "Use SWIG to generate bindings" ON)
+option (BUILD_CPP               "Build C++ interace (nlopt.hpp)" ON)
+option (BUILD_F77               "Build Fortran iterface" ON)
+option (BUILD_PYTHON            "Build Python bindings (using SWIG)" ON)
+option (BUILD_GUILE             "Build Guile bindings (using SWIG)" ON)
+option (BUILD_OCTAVE            "Build Octave bindings" ON)
+option (BUILD_MATLAB            "Build MATLAB bindings" ON)
+option (BUILD_TESTS             "Build tests" ON)
+option (BUILD_TESTS_UTIL        "Build util tests" OFF)
+option (BUILD_TESTS_ALG         "Build algorithms tests" OFF)
```

Comment:
noted, I'll remove the extra options (BUILD_TESTS* and BUILD_CPP).

I would argue to keep the Fortran one `BUILD_F77`, since it does involve exporting new symbols in the library, and there are valid cases where one would prefer a smaller executable/symbol-table if Fortran-compatible API is not being used.